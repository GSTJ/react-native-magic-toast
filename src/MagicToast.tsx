import React, { useEffect } from 'react';
import { StatusBar, Text, View } from 'react-native';
import { magicModal } from 'react-native-magic-modal';
import { useSafeAreaInsets } from 'react-native-safe-area-context';
import { AlertIcon } from './assets/AlertIcon';
import { styles } from './MagicToast.styles';

export const TOAST_TEST_ID = 'magic-toast';

export interface IMagicToast {
  message: string;
  duration?: number;
}

export const MagicToast: React.FC<IMagicToast> = ({
  message,
  duration = 3000,
}) => {
  const { top } = useSafeAreaInsets();

  useEffect(() => {
    const timeout = setTimeout(magicModal.hide, duration);
    return () => clearTimeout(timeout);
  }, [duration]);

  return (
    <View
      testID={TOAST_TEST_ID}
      style={[styles.container, { paddingTop: 25 + top }]}
    >
      <StatusBar barStyle="light-content" />
      <AlertIcon fill="white" width={20} height={20} style={styles.icon} />
      <Text style={styles.message}>{message}</Text>
    </View>
  );
};
