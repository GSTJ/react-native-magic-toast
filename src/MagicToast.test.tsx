import React from 'react';
import { render, act } from '@testing-library/react-native';
import { MagicModalPortal } from 'react-native-magic-modal';
import { magicToast } from './index';
import { TOAST_TEST_ID } from './MagicToast';
import { SafeAreaProvider } from 'react-native-safe-area-context';

describe('MagicToast', () => {
  it('renders correctly', async () => {
    const component = render(
      <SafeAreaProvider>
        <MagicModalPortal />
      </SafeAreaProvider>
    );

    expect(component.queryByTestId(TOAST_TEST_ID)).toBeFalsy();

    act(() => {
      magicToast.alert('Taveira');
    });

    expect(component).toMatchSnapshot();
    expect(component.queryByTestId(TOAST_TEST_ID)).toBeTruthy();
  });
});
