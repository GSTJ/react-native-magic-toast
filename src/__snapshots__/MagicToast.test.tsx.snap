// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MagicToast renders correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Array [
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
      Object {
        "backgroundColor": "transparent",
        "opacity": 1,
        "zIndex": 2,
      },
    ]
  }
>
  <View
    deviceHeight={1334}
    deviceWidth={750}
    forwardedRef={[Function]}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    onSwipeComplete={[Function]}
    panResponderThreshold={4}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={true}
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-start",
        "left": 0,
        "margin": 0,
        "opacity": 0,
        "top": 0,
        "transform": Array [
          Object {
            "translateY": -100,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeDirection="up"
    swipeThreshold={100}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#191919",
            "flexDirection": "row",
            "padding": 25,
          },
          Object {
            "paddingTop": 25,
          },
        ]
      }
      testID="magic-toast"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={20}
        bbWidth={20}
        fill="white"
        focusable={false}
        height={20}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "marginRight": 15,
            },
            Object {
              "flex": 0,
              "height": 20,
              "width": 20,
            },
          ]
        }
        vbHeight={512}
        vbWidth={512}
        width={20}
        x="0px"
        y="0px"
      >
        <RNSVGGroup
          fill={4294967295}
          propList={
            Array [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M501.362 383.95L320.497 51.474c-29.059-48.921-99.896-48.986-128.994 0L10.647 383.95c-29.706 49.989 6.259 113.291 64.482 113.291h361.736c58.174 0 94.203-63.251 64.497-113.291zM256 437.241c-16.538 0-30-13.462-30-30s13.462-30 30-30 30 13.462 30 30-13.462 30-30 30zm30-120c0 16.538-13.462 30-30 30s-30-13.462-30-30v-150c0-16.538 13.462-30 30-30s30 13.462 30 30v150z"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <Text
        style={
          Object {
            "color": "white",
            "fontWeight": "bold",
          }
        }
      >
        Taveira
      </Text>
    </View>
  </View>
</View>
`;
